tests/chainer_tests/functions_tests/rnn_tests/test_function_lstm.py

```diff
@@ -29,151 +28,112 @@ def inject_backend_tests(method_names):
     return decorator
 
 
-@testing.parameterize(*(testing.product({
-    'batch': [3, 2, 0],
-    'dtype': [numpy.float32],
-}) + testing.product({
-    'batch': [3],
-    'dtype': [numpy.float16, numpy.float32, numpy.float64],
-})))
+@testing.parameterize(*testing.product_dict(
+    [
+        {'c_shape': (10, 3), 'x_shape': (10, 12)},
+        {'c_shape': (20, 32, 4), 'x_shape': (16, 128, 4)},
+        {'c_shape': (32, 100, 3, 5), 'x_shape': (32, 400, 3, 5)},
+        {'c_shape': (16, 20), 'x_shape': (2, 80)},
```

Comment:
Could you add test cases: `x.size > 0`?

Reply:
I didn't understand. Please, could you give an example of the test cases I should add?

Reply:
`x.size == 0` is correct. Sorry for my typo.